project/dexter-core/src/java/com/samsung/sec/dexter/core/config/DexterConfigFile.java

```diff
@@ -42,457 +34,542 @@
+        IAnalysisEntityFactory configFactory = new AnalysisEntityFactory();
+        final AnalysisConfig analysisConfig = configFactory.createAnalysisConfig();
+
+        analysisConfig.setProjectName(getProjectName());
+        analysisConfig.setProjectFullPath(getProjectFullPath() + "/");
+        analysisConfig.setOutputDir(getBinDir());
+        analysisConfig.setModulePath(getModulePath());
+        analysisConfig.setSourceBaseDirList(getSourceDirList());
+        analysisConfig.setHeaderBaseDirList(getHeaderDirList());
+        analysisConfig.setLibDirList(getLibDirList());
+        analysisConfig.setSnapshotId(generateSnapshotId());
+        analysisConfig.setResultFileFullPath(getResultFileFullPath());
+        analysisConfig.setFunctionList(getFunctionList());
+        analysisConfig.setNoDefectGroupAndSnapshotId();
+
+        return analysisConfig;
+    }
+
+    @Override
+    public void setFields(final Map<String, Object> params) {
+        checkDexterConfigMap(params);
+
+        setDexterHome((String) params.get("dexterHome"));
+        setDexterServerIp((String) params.get("dexterServerIp"));
+        setDexterServerPort(DexterUtil.getIntFromMap(params, "dexterServerPort"));
+
+        setProjectName((String) params.get(ResultFileConstant.PROJECT_NAME));
+        setProjectFullPath((String) params.get("projectFullPath") + "/");
+        setSourceDirList(getStringListFromMap(params, "sourceDir"));
+        setHeaderDirList(getStringListFromMap(params, "headerDir"));
+        setSourceEncoding((String) params.get("sourceEncoding"));
+        setLibDirList(getStringListFromMap(params, "libDir"));
+        setBinDir((String) params.get("binDir"));
+        setType((String) params.get("type"));
+        setModulePath((String) params.get(ResultFileConstant.MODULE_PATH));
+        setFileNameList(getStringListFromMap(params, ResultFileConstant.FILE_NAME));
+        setResultFileFullPath((String) params.get("resultFileFullPath"));
+        setFunctionList(getStringListFromMap(params, "functionList"));
+        setSnapshotId((String) params.get(ResultFileConstant.SNAPSHOT_ID));
+        setDexterServerIp((String) params.get("dexterServerIp"));
+    }
+
+    protected void setType(String value) {
+        if (Type.hasValue(value)) {
+            this.type = Type.valueOf(value);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    protected List<String> getStringListFromMap(final Map<String, Object> map, String key) {
+        if (null == map.get(key) || (map.get(key) instanceof ArrayList) == false) {
+            return new ArrayList<String>(0);
+        }
+
+        return (ArrayList<String>) map.get(key);
+    }
+
+    private long generateSnapshotId() {
+        if ("SNAPSHOT".equals(type.toString()) == false) {
+            return -1;
+        }
+
+        if (Strings.isNullOrEmpty(this.snapshotId)) {
+            return System.currentTimeMillis();
+        } else {
+            return Integer.parseInt(this.snapshotId);
```

Comment:
snapshotId can take on 'Long' value. 
But It seems that return 'Int' value in this function.